### DISCUSSION TITLE
Unable to Register Group License on Customer's Computer without Docker

### DISCUSSION THREAD

**Discussion by acalicengiz** [dashingsoft/pyarmor#1403] [General]
Hello,

I purchased a pyarmor group license for the first time and activated it on my virtual Windows machine without any issues. I even accidentally activated the group license on my virtual machine. So everything works smoothly on the Windows virtual machine.

Now, I want to register the group license on the customer's computer, but despite hours of effort, I couldn't manage to do it. Is Docker necessary for Windows? I was able to activate the group license on my virtual machine without using Docker, and I have read your documentation at least 10 times and followed each step, but it still gives an error on the customer's computer. Thanks

**Log:**

```
ERROR could not get docker host machine id
Traceback (most recent call last):
File "C:\Users\User\AppData\Roaming\Python\Python311\site-packages\pyarmor\cli\register.py", line 183, in _get_docker_hostname
s.connect((host, port))
socket.gaierror: [Errno 11001] getaddrinfo failed
ERROR could not get docker host machine id
```

**Comment by jondy** [comment]
First of all, according to Pyarmor EULA, it's not allowed to register Pyarmor License on customer's computer.

Pyarmor is used to obfuscate your own scripts, why register it in customer's computer?

**Reply by acalicengiz** [reply]
I think we misunderstood you. Is the correct as follows:

Sample Product Name: Accounting.py

1. We encrypt our product on our own computer using pyarmor.

2. We install the trial-licensed pyarmor on the customer's computer using "pip install pyarmor."

3. We transfer the encrypted "Accounting.py" file to the customer's computer.

4. Additionally, since "Accounting.py" is encrypted by a licensed pyarmor, we do not require any further license installations on the customer's computer. Pyarmor detects that it has been encrypted by a licensed user.

5. It doesn't matter if pyarmor on the customer's computer is in trial mode.

Is this correct?

**Comment by joonis** [comment]
You don't need to install Pyarmor on the customer's computer at all.
